### PR TITLE
Improve processing step

### PR DESCRIPTION
**Pull request by babgev**
- Improve processing step
- Add storybook integration
- Replace in-house error boundary with react-error-boundary package
- Improve resource storage URL resolution

### CODE REVIEW

File: Models/LayerSwapAppSettings.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { LayerSwapSettings } from "./LayerSwapSettings";
 import { Partner } from "./Partner";
 
 export class LayerSwapAppSettings extends LayerSwapSettings {
-    constructor(settings: LayerSwapSettings) {
+    constructor(settings: LayerSwapSettings | any) {
```
Comment (babgev) [inline Models/LayerSwapAppSettings.ts:9]: Why the added any type?

File: Models/Wizard.ts [hunk 2]
```diff
@@ -32,17 +32,17 @@ export enum SwapCreateStep {
     Error = "Error"
 }
 
-export enum SwapStep {
-    UserTransferPending,
-    
-    TransactionDone,
-    TransactionDetected,
-    LSTransferPending,
+// export enum SwapStep {
```
Comment (babgev) [inline Models/Wizard.ts:35]: Delete instead of commenting

File: components/layout.tsx [hunk 3]
```diff
@@ -91,7 +90,7 @@ export default function Layout({ hideNavbar, children }: Props) {
     {
       storageAvailable === true &&
       <AuthProvider>
-        <ErrorBoundary >
+        <ErrorBoundary FallbackComponent={ErrorFallback}>
```
Comment (babgev) [inline components/layout.tsx:93]: Need to test this error boundry

File: components/modal/modal.tsx [hunk 4]
```diff
@@ -40,8 +40,8 @@ const Modal: FC<ModalProps> = (({ header, height, className, children, subHeader
                             </ReactPortal>
                         }
                         {
-                            isMobile &&
-                            <Leaflet position="fixed" height={height == 'full' ? '80%' : 'full'} ref={mobileModalRef} show={show} setShow={setShow} title={header} description={subHeader} className={className}>
+                        isMobile &&
+                            <Leaflet position="fixed" height={height == 'full' ? '80%' : height == 'fit' ? 'fit' : 'full'} ref={mobileModalRef} show={show} setShow={setShow} title={header} description={subHeader} className={className}>
```
Comment (babgev) [inline components/modal/modal.tsx:44]: What's this change?

File: components/utils/SwapStatus.ts [hunk 5]
```diff
@@ -1,36 +1,34 @@
...
-        return SwapStep.TransactionDone;
-    else
-        return SwapStep.UserTransferPending
-}
+//     if (swapStatus == SwapStatus.Completed)
+//         return SwapStep.Success;
+//     else if (swapStatus == SwapStatus.Failed || swapStatus == SwapStatus.Cancelled || swapStatus === SwapStatus.Expired)
+//         return SwapStep.Failed;
+//     else if (swapStatus == SwapStatus.UserTransferDelayed)
+//         return SwapStep.Delay;
```
Comment (babgev) [inline components/utils/SwapStatus.ts:16]: Delete instead of commenting

File: pages/campaigns/index.tsx [hunk 6]
```diff
@@ -36,13 +36,8 @@ export async function getServerSideProps(context) {
 
     settings.exchanges = settings.exchanges.filter(e => e.status === 'active')
 
-    const resource_storage_url = settings.discovery.resource_storage_url
-    if (resource_storage_url[resource_storage_url.length - 1] === "/")
-        settings.discovery.resource_storage_url = resource_storage_url.slice(0, -1)
-
     let isOfframpEnabled = process.env.OFFRAMP_ENABLED != undefined && process.env.OFFRAMP_ENABLED == "true";
```
Comment (babgev) [inline pages/campaigns/index.tsx:39]: - Remove this line, not used anymore
- Create a function that can be reused in every getserverside props to fetch the settings

File: Models/LayerSwapAppSettings.ts (hunk 1 above)
Comment (paur94) [reply Models/LayerSwapAppSettings.ts:9]: fixed

File: components/Swap/StepsComponent.tsx [hunk 7]
```diff
@@ -43,20 +43,50 @@ export default function Steps({ steps }) {
...
+                                        <span className="flex h-9 items-center" aria-hidden="true">
+                                            <XCircle className="h-8 w-8 text-red-600" aria-hidden="true" />
+                                        </span>
+                                        <span className="ml-4 flex min-w-0 flex-col">
+                                        <span className="text-sm font-medium text-gray-300">{step.name}</span>
+                                            <span className="text-sm text-primary-text">{step?.description}</span>
+                                        </span>
+                                    </div>
+                                </>
+                            ) : step?.status === 'user_transfer_delayed' ? (
```
Comment (babgev) [inline components/Swap/StepsComponent.tsx:61]: Don't have a check on `user_transfer_delayed`. Add a more generic term like `delayed` that can later also be used for other cases + will decouple this component from swap's status.

File: .storybook/preview.ts [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+
+export const parameters = {
+  actions: { argTypesRegex: "^on[A-Z].*" },
+};
+
+const preview: Preview = {
+  // decorators: [
+  //   (Story) => {
+  //     window.localStorage.clear();
```
Comment (babgev) [inline .storybook/preview.ts:14]: Remove comment

File: components/Swap/StepsComponent.tsx [hunk 9]
```diff
@@ -43,20 +46,51 @@ export default function Steps({ steps }) {
...
+                                    {stepIdx !== filteredSteps.length - 1 ? (
+                                        <div className="absolute top-1/2 left-4 -ml-px mt-0.5 h-[40%] w-0.5 bg-gray-300 opacity-60" aria-hidden="true" />
+                                    ) : null}
+                                    <div className={`group relative flex ${step?.description ? "items-start" : "items-center"}`}>
+                                        <span className="flex h-9 items-center" aria-hidden="true">
+                                            <XCircle className="h-8 w-8 text-red-600" aria-hidden="true" />
+                                        </span>
+                                        <span className="ml-4 flex min-w-0 flex-col">
+                                            <span className="text-sm font-medium text-gray-300">{step.name}</span>
+                                            {step?.description &&
```
Comment (babgev) [inline components/Swap/StepsComponent.tsx:60]: These steps have the same code, we can extract it into a Step component and reuse

**Pull request merged**